src/sexpr_tokenizer.rs

```diff
@@ -0,0 +1,199 @@
+use std::iter::Peekable;
+use std::str::Chars;
+
+/// The representation of an s-expression parse error.
+#[derive(Debug)]
+pub struct TokenizerError {
+    /// The error message.
+    pub message: &'static str,
+    /// The line number on which the error occurred.
+    pub line: usize,
+    /// The column number on which the error occurred.
+    pub column: usize,
+}
+
+#[derive(PartialEq, Clone, Debug)]
+pub enum Token {
+    Word(String),
+    /// Word that starts with a `'`
+    Atom(String),
+    Number(String),
```

Comment:
No, because we just pass the string typed of number literal to the parser. At this moment the tokenizer would not know this is `Atom::I` or `Atom::F`.

Reply:
Fair enough

Reply:
Fair enough